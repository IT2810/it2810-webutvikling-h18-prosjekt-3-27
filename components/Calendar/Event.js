import React, {PureComponent} from "react";
import {
  Text,
  StyleSheet, TouchableOpacity
} from "react-native";

export default class Event extends PureComponent {

  handleEventPress = () => {
    if (this.props.event && this.props.onEventPress) {
      this.props.onEventPress(this.props.event);
    }
  };

  render() {
    let eventName = "";
    let eventNote = "";
    if (this.props.event) {
      eventName = this.props.event.name;
      eventNote = this.props.event.note;
      const maxNoteLength = 35;
      if (eventNote.length > maxNoteLength) {
        eventNote = eventNote.slice(0, maxNoteLength - 4) + "...";
      }
    }
    return (
      <TouchableOpacity
        style={styles.item}
        onPress={this.handleEventPress}
      >
        <Text>{eventName}</Text>
        <Text style={styles.note}>{eventNote}</Text>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  item: {
    backgroundColor: "white",
    flex: 1,
    borderRadius: 5,
    padding: 10,
    marginRight: 10,
    marginTop: 17
  },
  note: {
    color: "#828282"
  }
});
