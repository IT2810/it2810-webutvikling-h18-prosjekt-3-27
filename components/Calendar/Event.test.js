import React from "react";
import renderer from "react-test-renderer";
import Event from "./Event";

it("renders correctly", () => {
  const tree = renderer
    .create(<Event/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
