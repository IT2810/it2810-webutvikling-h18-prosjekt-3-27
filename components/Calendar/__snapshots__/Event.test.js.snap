// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "flex": 1,
      "marginRight": 10,
      "marginTop": 17,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#828282",
      }
    }
  >
    
  </Text>
</View>
`;

exports[`renders events correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "flex": 1,
      "marginRight": 10,
      "marginTop": 17,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    The event name
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#828282",
      }
    }
  >
    A longer description which is s...
  </Text>
</View>
`;
